modules/create_people.rb

```diff
@@ -0,0 +1,67 @@
+module CreatePeople
+  def permission_given?
+    parent_permission = true
+    loop do
+      option = gets.chomp.upcase!
+      case option
+      when 'Y'
+        break
+      when 'N'
+        parent_permission = false
+        break
+      else
+        puts 'Please enter a valid option'
+      end
+    end
+    parent_permission
+  end
+
+  def select_role
+    loop do
+      puts 'Enter 1 for Student or 2 for Teacher'
+      person_opt = gets.chomp
+      case person_opt
+      when '1'
+        create_student
+        break
+      when '2'
+        create_teacher
+        break
+      else
+        puts 'Invalid option, please try again'
+      end
```

Comment:
- [ ] As pointed out above, all requests for input and input validations should be done somewhere else whose responsibility is to request for inputs or validate user inputs.

Reply:
@FrederickMih  Absolutely noone separates these two things.